src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -509,10 +552,19 @@ private Type EmitIdentifierListNode(AssociativeNode node)
             var iln = node as IdentifierListNode;
             if(iln == null) throw new ArgumentException("AST node must be an Identifier List.");
 
-            var ident = iln.LeftNode as IdentifierNode;
-            if (ident == null) throw new ArgumentException("Left node of IdentifierListNode is expected to be an identifier.");
+            //TODO this is a bit of a hack to support class names like DSCore.Math
+            //only works one level deep, I guess this should recurse?
+            if (iln.LeftNode is IdentifierListNode)
+            {
+                className = iln.LeftNode.ToString();
+            }
+            else
+            {
+                var ident = iln.LeftNode as IdentifierNode;
+                if (ident == null) throw new ArgumentException("Left node of IdentifierListNode is expected to be an identifier.");
 
-            className = ident.Value;
+                className = ident.Value;
+            }
```

Comment:
I think this entire logic seems flaky. We should be able to use this existing utility method here, `CoreUtils.GetIdentifierExceptMethodName(IdentifierListNode identList)`. Can you try that?

Reply:
https://github.com/DynamoDS/Dynamo/blob/e5e6d90511ab8ea84bd45d193071c3006e96c1cb/src/Engine/ProtoCore/Utils/CoreUtils.cs#L518